pinecone/core/grpc/index_grpc.py

```diff
@@ -171,9 +171,10 @@ def parse_fetch_response(response: dict):
     return FetchResponse(vectors=vd, namespace=namespace, _check_type=False)
 
 
-def parse_query_response(response: dict, unary_query: bool):
+def parse_query_response(response: dict, unary_query: bool, _check_type: bool = False):
```

Comment:
I suggest using a similar mechanism as in on the `Index` object where we use kwargs and propagate it this way

Reply:
On the contrary. In `Index` we remove this param from the `kwargs` and enforce our own default value, then explicitly pass it to downstream functions. See for example [here](https://github.com/pinecone-io/pinecone-python-client/blob/main/pinecone/index.py#L97).  
That's why I used the same mechanism here.

(Remember that the top-level function is `query()`, not `parse_response()`).

Reply:
Really, I'd actually just hardcode it False w/o enabling users to override; since data is exchanged between client and server as protobuf, the main _check_type usecase of type-checking to detect out-of-sync client & server will already be handled by the protobuf ser/de. Doing _check_type here only checks our grpc->oapi conversion logic, which would be better tested by unit tests. And if we expose the functionality we'll want to document it, and if a user sees that they have an option to "check types" they'll probably do it....

Reply:
@benjaminran makes sense.  
Ok, changing it now.

Reply:
1005